app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
Create an unmanaged host via hammer and change it to managed (skips validations). Things like Importing a host via facts skip validations as well. I guess the later can't cause this issue but definitely convinces me to be defensive.